script.js

```diff
@@ -1,4 +1,184 @@
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+var deck = makeDeck();
+var shuffledDeck = shuffleCards(deck);
+
+
+//function to take a hand and come up with points
+//add rank
+// count aces
+// write the score
+var calculate = function(hand){
+  var currentScore = 0;
+  var cardIndex = 0;
+  var handLength = hand.length;
+  var gotAce = 0;
+  while (cardIndex < handLength){
+    var currentCard = hand[cardIndex]
+    var cardScore = currentCard.rank
+    if (cardScore == 1){
+      gotAce += 1;
+    }
+    currentScore += Math.min(10,cardScore)
+    cardIndex += 1
+  }
+  if (gotAce > 0 & currentScore <= 11){
+    currentScore += 10;
+  }
+  return currentScore
+}
+
+//banker to decide whether to draw or not
+var dealerActions = function(){
+  var currentScore = calculate(computerHand);
+  console.log(currentScore)
+  while (currentScore < 17){
+    var newCard = shuffledDeck.pop();
+    computerHand.push(newCard);
+    currentScore = calculate(computerHand);
+    console.log(currentScore)
+  }
+}
+
+
+// if user choose stand, compare card with computer
+// show results and end game.
+
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // Submit to deal 2 cards to computer and player
+  if (gameMode == 'deal2Cards'){
+    var playerCard1 = shuffledDeck.pop();
+    var playerCard2 = shuffledDeck.pop();
+    playerHand.push(playerCard1,playerCard2);
+    var computerCard1 = shuffledDeck.pop();
+    var computerCard2 = shuffledDeck.pop();
+    computerHand.push(computerCard1,computerCard2);
+
+
+    //blackjack check
+    if (calculate(playerHand)== 21){
+      var myOutputValue = 'The cards are dealt. You have drawn ' +playerCard1.name+ ' of ' +playerCard1.suit+ ' and ' +playerCard2.name+ ' of ' +playerCard2.suit + '<br><b>Congratulations, you have hit Blackjack</b>';
+      playerHand = [];
+      deck = makeDeck();
+      shuffledDeck = shuffleCards(deck);
+      return myOutputValue;
+    }
+    // normal output
+    var myOutputValue = 'The cards are dealt. You have drawn ' +playerCard1.name+ ' of ' +playerCard1.suit+ ' and ' +playerCard2.name+ ' of ' +playerCard2.suit + ' Choose "hit" or "stand" for your next move!';
+    gameMode = 'choose Hit or Stand';
+    return myOutputValue;
+    
+  } else if (gameMode == 'choose Hit or Stand'){
+      // Player choose hit
+      if (input == 'hit'){
+        var playerCard = shuffledDeck.pop();
+        playerHand.push(playerCard);
+        var playerScore = calculate(playerHand);
+
+        // Display cards
+        var myOutputValue = "These are your cards:<br>";
+        var cardIndex = 0;
+        var handLength = playerHand.length;
+        while (cardIndex < handLength){
+          var currentCard = playerHand[cardIndex];
+          myOutputValue += `${currentCard.name} of ${currentCard.suit} | `;
+          cardIndex += 1;
+        }
```

Comment:
A helper function for this would be ideal too! I think displaying cards is something that you might re-use multiple times throughout the game